From d57ef3fa5b95e3f040134bddb131ca6483a76486 Mon Sep 17 00:00:00 2001
From: Janne Kajovuori <janne.kajovuori@nokia.com>
Date: Mon, 27 May 2019 06:23:04 +0000
Subject: [PATCH 0915/1239] octeontx2: decouple arch and board configurations

Enable target board selection for mach-octeontx2 instead of using one
architecture specific configuration for all boards. This allows U-Boot
to be extended with additional vendor boards that use OCTEON TX2 SoC.

The existing virtual 'octeontx2' board, which supports multiple
reference boards, is still supported as before when the build is
configured for Marvell CN95XX and CN96XX targets.

Change-Id: Iff2773afa5ab30d9d39c21fd6bbc05aa6ccb31c0
Signed-off-by: Janne Kajovuori <janne.kajovuori@nokia.com>
Reviewed-on: https://sj1git1.cavium.com/10366
Reviewed-by: Suneel Garapati <sgarapati@caviumnetworks.com>
Tested-by: Suneel Garapati <sgarapati@caviumnetworks.com>
---
 arch/arm/mach-octeontx2/Kconfig | 18 +++++++++++-------
 board/Marvell/octeontx2/Kconfig | 17 +++++++----------
 2 files changed, 18 insertions(+), 17 deletions(-)

diff --git a/arch/arm/mach-octeontx2/Kconfig b/arch/arm/mach-octeontx2/Kconfig
index 9605ddd5c3..8e5cb0f638 100644
--- a/arch/arm/mach-octeontx2/Kconfig
+++ b/arch/arm/mach-octeontx2/Kconfig
@@ -1,14 +1,18 @@
 if ARCH_OCTEONTX2
 
-config SYS_VENDOR
-	string
-	default	"Marvell"
+choice
+	prompt "OcteonTX2 board select"
+	optional
 
-config SYS_SOC
-	string
-	default "octeontx2"
+config TARGET_OCTEONTX2_95XX
+	bool "Marvell OcteonTX2 CN95XX"
 
-config SYS_BOARD
+config TARGET_OCTEONTX2_96XX
+	bool "Marvell OcteonTX2 CN96XX"
+
+endchoice
+
+config SYS_SOC
 	string
 	default "octeontx2"
 
diff --git a/board/Marvell/octeontx2/Kconfig b/board/Marvell/octeontx2/Kconfig
index 793340b6af..e298b4cc85 100644
--- a/board/Marvell/octeontx2/Kconfig
+++ b/board/Marvell/octeontx2/Kconfig
@@ -1,15 +1,12 @@
-if ARCH_OCTEONTX2
+if TARGET_OCTEONTX2_95XX || TARGET_OCTEONTX2_96XX
 
-choice
-	prompt "Target board"
+config SYS_VENDOR
+	string
+	default	"Marvell"
 
-config TARGET_OCTEONTX2_96XX
-	bool "OcteonTX2 CN96XX"
-
-config TARGET_OCTEONTX2_95XX
-	bool "OcteonTX2 CN95XX"
-
-endchoice
+config SYS_BOARD
+	string
+	default "octeontx2"
 
 config SYS_CONFIG_NAME
 	default "octeontx2_96xx" if TARGET_OCTEONTX2_96XX
-- 
2.29.0

